lib/hipcore_typed/rewrite_context.ml

```diff
@@ -0,0 +1,106 @@
+
+open Hipcore
+open Typedhip
+open Common
+
+type map_context = {
+  (* list of bound names *)
+  mctx_bound: SSet.t;
+  (* list of renames needed for capture avoidance.
+     these are lazy values so that the new variable name is only generated if needed *)
+  mctx_captured_renames: string Lazy.t SMap.t;
+}
```

Comment:
Maybe add an mli for this module which shows what the API is? For example, can this type be left abstract?

From a brief look, it seems you're supposed to define a visitor inheriting from map_spec_with_binders, call it using visit_using_env, then use is_bound inside it?